src/test/scala/com/microsoft/hyperspace/index/HybridScanForPartitionedDataSuite.scala

```diff
@@ -27,78 +28,34 @@ import com.microsoft.hyperspace.util.FileUtils
 class HybridScanForPartitionedDataSuite extends HybridScanTestSuite {
   override def beforeAll(): Unit = {
     super.beforeAll()
-    import spark.implicits._
-    val dfFromSample = sampleData.toDF("Date", "RGUID", "Query", "imprs", "clicks")
     hyperspace = new Hyperspace(spark)
+  }
 
-    FileUtils.delete(new Path(sampleDataLocationRoot))
-    dfFromSample.write
-      .partitionBy("clicks", "Date")
-      .format(fileFormat)
-      .save(sampleDataFormatAppend)
-    dfFromSample.write.partitionBy("Date").format(fileFormat).save(sampleDataFormatAppend2)
-    dfFromSample.write.partitionBy("Date").format(fileFormat).save(sampleDataFormatDelete)
-    dfFromSample.write.partitionBy("imprs").format(fileFormat).save(sampleDataFormatDelete2)
-    dfFromSample.write
-      .partitionBy("Date", "imprs")
-      .format(fileFormat)
-      .save(sampleDataFormatDelete3)
-    dfFromSample.write.partitionBy("Date").format(fileFormat).save(sampleDataFormatDelete4)
-    dfFromSample.write.partitionBy("Date").format(fileFormat).save(sampleDataFormatBoth)
-    dfFromSample.write
-      .partitionBy("Date", "clicks")
-      .format(fileFormat2)
-      .save(sampleDataFormat2Append)
-    dfFromSample.write.partitionBy("Date").format(fileFormat2).save(sampleDataFormat2Delete)
-
-    val indexConfig1 = IndexConfig("indexType1", Seq("clicks"), Seq("query"))
-    val indexConfig2 = IndexConfig("indexType2", Seq("clicks"), Seq("Date"))
-
-    setupIndexAndChangeData(
-      fileFormat,
-      sampleDataFormatAppend,
-      indexConfig1.copy(indexName = "index_ParquetAppend"),
-      appendCnt = 1,
-      deleteCnt = 0)
-    setupIndexAndChangeData(
-      fileFormat,
-      sampleDataFormatAppend2,
-      indexConfig2.copy(indexName = "indexType2_ParquetAppend2"),
-      appendCnt = 1,
-      deleteCnt = 0)
-    setupIndexAndChangeData(
-      fileFormat2,
-      sampleDataFormat2Append,
-      indexConfig1.copy(indexName = "index_JsonAppend"),
-      appendCnt = 1,
-      deleteCnt = 0)
+  override def setupIndexAndChangeData(
+      sourceFileFormat: String,
+      sourcePath: String,
+      indexConfig: IndexConfig,
+      appendCnt: Int,
+      deleteCnt: Int): (Seq[String], Seq[String]) = {
+    dfFromSample.write.partitionBy("clicks", "Date").format(sourceFileFormat).save(sourcePath)
```

Comment:
Note: as this function needs to create source data accordingly, added (override) function for partitioned data.